Fenrir.Tests/CommandTests.fs

```diff
@@ -119,3 +118,117 @@ let ``Restoring head should work properly``(): unit =
+
+    Assert.Equal(treeStreams.Hashes.[0], parentHash)
+    Assert.Equal(treeStreams.Hashes.[1], subTreeHash)
+
+    let tr = Commands.streamToTreeBody treeStreams.Streams.[0]
+    let subTr = Commands.streamToTreeBody treeStreams.Streams.[1]
+
+    Assert.Equal(tr.Length, 2)
+    Assert.Equal(tr.[0].Mode, 100644UL)
+    Assert.Equal(tr.[0].Name, "README")
+    Assert.Equal<byte>(tr.[0].Hash, readmeHash |> Commands.stringToByte)
+    Assert.Equal(tr.[1].Mode, 40000UL)
+    Assert.Equal(tr.[1].Name, "ex")
+    Assert.Equal<byte>(tr.[1].Hash, subTreeHash |> Commands.stringToByte)
+
+    Assert.Equal(subTr.Length, 1)
+    Assert.Equal(subTr.[0].Mode, 100644UL)
+    Assert.Equal(subTr.[0].Name, "FIGHTTHEMACHINE")
+    Assert.Equal<byte>(subTr.[0].Hash, oldBlobHash |> Commands.stringToByte)
+
+[<Fact>]
+let ``updateObjectInTree should change the whole tree properly``(): unit =
+    let oldParentHash = "0ba2ef789f6245b6b6604f54706b1dce1d84907f"
+    let newParentHash = "a3ecc1b7fb40831db85596a4f6d2b5e0a1070292"
+    let newSubTreeHash = "b6c6d6bca44755db41e85040189d86c0dbec691e"
+    let newBlobHash = "724978a20d84133868886a8e580f59c6f8586733"
+    let readmeHash = "e2af08e76b2408a88f13d2c64ca89f2d03c98385"
+
+    let pathToFile = Path.Combine("ex", "FIGHTTHEMACHINE")
+    use treeStreams = Commands.updateObjectInTree oldParentHash testDataRoot pathToFile newBlobHash
+
+    Assert.Equal(treeStreams.Hashes.[0], newParentHash)
+    Assert.Equal(treeStreams.Hashes.[1], newSubTreeHash)
+
+    let tr = Commands.streamToTreeBody treeStreams.Streams.[0]
+    let subTr = Commands.streamToTreeBody treeStreams.Streams.[1]
+
+    Assert.Equal(tr.Length, 2)
+    Assert.Equal(tr.[0].Mode, 100644UL)
+    Assert.Equal(tr.[0].Name, "README")
+    Assert.Equal<byte>(tr.[0].Hash, readmeHash |> Commands.stringToByte)
+    Assert.Equal(tr.[1].Mode, 40000UL)
+    Assert.Equal(tr.[1].Name, "ex")
+    Assert.Equal<byte>(tr.[1].Hash, newSubTreeHash |> Commands.stringToByte)
+
+    Assert.Equal(subTr.Length, 1)
+    Assert.Equal(subTr.[0].Mode, 100644UL)
+    Assert.Equal(subTr.[0].Name, "FIGHTTHEMACHINE")
+    Assert.Equal<byte>(subTr.[0].Hash, newBlobHash |> Commands.stringToByte)
+
+[<Fact>]
+let ``Files should be written after updating of the whole tree``(): unit =
+    let oldParentHash = "0ba2ef789f6245b6b6604f54706b1dce1d84907f"
+    let newBlobHash = "724978a20d84133868886a8e580f59c6f8586733"
+    let pathToParentTree = Path.Combine(testDataRoot, ".git", "objects", "a3", "ecc1b7fb40831db85596a4f6d2b5e0a1070292")
+    let pathToSubTree = Path.Combine(testDataRoot, ".git", "objects", "b6", "c6d6bca44755db41e85040189d86c0dbec691e")
+
+    let pathToFile = Path.Combine("ex", "FIGHTTHEMACHINE")
+    use treeStreams = Commands.updateObjectInTree oldParentHash testDataRoot pathToFile newBlobHash
+    Commands.writeTreeObjects testDataRoot treeStreams
+    Assert.True(File.Exists(pathToParentTree))
+    Assert.True(File.Exists(pathToSubTree))
```

Comment:
I suggest to delete the created files after test completion.

If you don't do that, it's possible to end up in a situation when the files are _already_ in the `testDataRoot` directory, the corresponding code is broken (i.e. doesn't create the files), but the test still passes.

It won't affect our CI (since it always starts from a clean checkout), but it will break the test flow locally.